emails/utils.py

```diff
@@ -218,12 +219,17 @@ def get_post_data_from_request(request):
 
 
 def generate_relay_From(original_from_address, user_profile=None):
+    _, relay_from_address = parseaddr(settings.RELAY_FROM_ADDRESS)
+    try:
+        new_from_flag = Flag.objects.get(name='new_from_address')
+        if user_profile and new_from_flag.is_active_for_user(user_profile.user):
```

Comment:
Nit: test for flag exists but not for that user. And maybe flag exists but `user_profile=None`

Reply:
Finally got these tests working but I had to explicitly call `new_from_flag.users.remove(free_user)` at the end of this test too. I guess `pytest-django` doesn't clean up the `Flag` objects between tests the same way I expected?

Reply:
Ug, that one sounds hard to debug!

Looking at the [docs](https://waffle.readthedocs.io/en/stable/testing/automated.html#unit-tests), the supported method is the `override_flag` decorator, which potentially will clean up after itself, although I see a few [open issues](https://github.com/django-waffle/django-waffle/issues/344) about cache clearing as well.

It may be worth investigating if 1) the database row in the many-to-many table is deleted during test cleanup, and 2) if there is a cache entry that is created and 3) if the cache entry is cleaned up.